peer/x/roundrobin/list.go

```diff
@@ -120,20 +121,15 @@ func (pl *List) addToAvailablePeers(p peer.Peer) error {
 
 // Start notifies the List that requests will start coming
 func (pl *List) Start() error {
-	if pl.started.Swap(true) {
-		return peer.ErrPeerListAlreadyStarted("RoundRobinList")
-	}
-	return nil
+	return pl.startOnce.Do(func() error { return nil })
```

Comment:
Yeah, I’m feeling pretty strongly that this should be boring, tedious, repetitive, without abstraction, and fundamentally readable.